irods_testing_environment/test_runner.py

```diff
@@ -0,0 +1,130 @@
+        tests -- optional list of tests to use with this runner
+        """
+        # TODO: each test runner is tied to a single container... might need to abstract this out later
+        self.executor = executing_container
+        self.tests = tests or list()
+        #self.passed_tests = list()
+        self.failed_tests = list()
+        self.rc = 0
+
+
+    def __str__(self):
+        """Return a string representation of a map representing the data members."""
+        return str({
+            'executing_container': self.name(),
+            'return_code': self.rc,
+            'test_list': self.tests,
+            #'passed_tests': self.passed_tests,
+            'failed_tests': self.failed_tests
+        })
+
+
+    def add_test(self, test):
+        """Append `test` to the test list and return self."""
+        logging.info('before [{}]'.format(self))
+        self.tests.append(test)
+        logging.info('after  [{}]'.format(self))
+        return self
+
+
+    def name(self):
+        """Return the name of the executing container."""
+        return self.executor.name
+
+
+    def result_string(self):
+        """Return a string representing the results of running the test list."""
+        r = '-----\nresults for [{}]\n'.format(self.name())
+
+        r = r + '\ttest list:\n'
+        for t in self.tests:
+            r = r + '\t\t[{}]\n'.format(t)
+
+        #r = r + '\tpassed tests:\n'
+        #for f in self.passed_tests:
+            #r = r + '\t\t[{}]\n'.format(f)
+
+        r = r + '\tfailed tests:\n'
+        for f in self.failed_tests:
+            r = r + '\t\t[{}]\n'.format(f)
+
+        r = r + '\treturn code:[{}]\n-----\n'.format(self.rc)
+
+        return r
+
+
+    def run(self, fail_fast=True, *args):
+        """Execute test list sequentially on executing container.
+
+        Arguments:
+        fail_fast -- if True, the first test to fail ends the run
+        *args -- any additional arguments that the test execution can take
+        """
+        for t in self.tests:
+            cmd, ec = self.execute_test(t, *args)
+
+            if ec is 0:
+                #self.passed_tests.append(t)
+                logging.debug('[{}]: cmd succeeded [{}]'.format(self.name(), cmd))
+
+            else:
+                self.rc = ec
+                self.failed_tests.append(t)
+                logging.warning('[{}]: cmd failed [{}] [{}]'.format(self.name(), ec, cmd))
+
+                if fail_fast:
+                    raise RuntimeError('[{}]: command failed [{}]'.format(self.name(), cmd))
+
+        if self.rc is not 0:
+            logging.error('[{}]: tests that failed [{}]'.format(self.name(), self.failed_tests))
+
+
+    @staticmethod
+    def run_tests_command():
+        """Return a list of strings used as a space-delimited invocation of the test runner."""
+        raise NotImplementedError('test_runner is a base class and should not be used directly')
```

Comment:
This probably doesn't need to be part of the interface as it is only used internally at this point

Reply:
What/How was it intended to be used?

Is it unnecessary now?

Reply:
Right, it's not necessary anymore. `test_runner_irods_python_suite` uses the method to abstract the command invocation for running the tests, but it's not required anymore because `execute_test` now returns the string of the command executed in a tuple with the return code.